### DISCUSSION TITLE
Document update conflict [Client] HTTP 400 on POST error - what does it mean?

### DISCUSSION THREAD

**Discussion by xlog-null** [Budibase/budibase#8093] [Help] [answered]
I have a page with a form that loads some rows from a MySQL database into list fields. Every time I load the page the red error shows up on the top: 
![image](https://user-images.githubusercontent.com/92792093/193525850-33723d88-031d-46fd-b388-dc7a956e1971.png)

And this is what shows up in the web browser console:
![image](https://user-images.githubusercontent.com/92792093/193526254-5b4b0564-0e89-4199-9920-1e84e62f06a3.png)

Now, most of the times the values are being loaded but about 20% of the time the fields that are supposed to hold the values from the database are empty and I need to refresh a few times for them to show up.

Any ideas?

**Comment by melohagan** [comment]
Hey @xlog-null 

Would you be able to share your app export at all? Just out of curiosity, can you try re-fetching your MySQL tables, and re-adding the data providers to see if that helps?

**Comment by xlog-null** [comment]
Hi, thanks for the response. I'm attaching my exported app.

I tried to re-fetch MySQL tables and re-add the data providers - no change at all.

I have noticed a pattern to it though. When I launch the app in a new tab the document conflict error doesn't show up and the fields are loaded properly. However after I refresh a few times, the conflict happens the majority of the time. When I close the tab and open the app again it's gone again until a couple of refreshes.

Another thing about that is that it's always two fields, two MySQL query requests that are failing. I have 4 fields in total on my form from 4 different tables, and every time the document conflict error happens, it happens twice, for two different fields. It seems to choose random fields/tables to fail on every time.

I am using the free Budibase version - there is no way that would affect this right? The tables fetched are quite small, around megabyte in total.

I'm losing hair and sleep over this... Thank you!

[app_export.txt](https://github.com/Budibase/budibase/files/9776177/Corrective.and.Preventative.Actions-export-1665654739012.txt)

**Reply by melohagan** [reply]
Hey @xlog-null What is the name of the screen that you are seeing the issues on? 

> Another thing about that is that it's always two fields, two MySQL query requests that are failing. I have 4 fields in total on my form from 4 different tables, and every time the document conflict error happens, it happens twice, for two different fields. It seems to choose random fields/tables to fail on every time.

Could you elaborate a bit more on that please. Which two fields? Which MySQL query requests are failing?

**Reply by xlog-null** [reply]
@melohagan I have two screens (almost the same) that contain the same fields including the four fields I am having issues with. The screens are `new` and `new_ofi`.

> Could you elaborate a bit more on that please. Which two fields? Which MySQL query requests are failing?

I have four fields that are Option Pickers. They are all linked to their corresponding Data Source Providers which are linked to their corresponding tables in the MySQL database. The option picker fields are: `source_picker`, `client_picker`, `project_picker` and `staff_picker`.

When I enter the app and the screens the first time - no problem. But when I refresh the page, most of the time I get the Document Update Conflict. When I check the web browser console, 90% of the time two requests to fetch corresponding table data for the Options Picker fail. But sometimes only one fails, I have just noticed.

![image](https://user-images.githubusercontent.com/92792093/195623083-25899d52-17f6-4cb7-9202-83580cf644fe.png)
The tables data requests that fail seem to be random in four tables - sometimes it's the source table, sometimes the customer table, sometimes the staff and sometimes the project table.

Hopefully that clarifies it!

**Reply by melohagan** [reply]
Where you see the red error, can you expand that carrot. Is there any further detail on what the 400 is, or is it simply document update conflict?

**Reply by melohagan** [reply]
I noticed as well that you have limits of **99999** on your options picker data providers. Just how many options are you expecting?

**Reply by xlog-null** [reply]
@melohagan 

![image](https://user-images.githubusercontent.com/92792093/195630031-0c067ab0-3768-499e-b953-4fd8c1f173c1.png)

This is what I get when expanding, it goes all the way down like this, it's the same for the orange error too. Doesn't give much info...

The limits were 99999 to be safe however it was a silly assumption as I don't expect more than 2k rows for the largest table in the future. However when changing the fields limit to 2000, the errors and behaviour is exactly the same.

**Reply by melohagan** [reply]
Your version of Budibase is pretty old. Can you upgrade to **v2.0.30** ?

**Reply by xlog-null** [reply]
I will update it as soon as regain access to our Docker droplet on which it's running, I will get back to you. I appreciate the help so far, thanks.

**Comment by xlog-null** [comment, marked as answer]
@melohagan
After creating a new droplet that allowed me to set up Budibase with version 2, my problem has been fixed. No more Document update conflict and the fields are loading from MySQL just fine! Thanks for your time.
